python/src/aac/plugins/validators/__init__.py

```diff
@@ -7,14 +7,16 @@
 from aac.plugins.validators._common import (
     get_validation_definition_from_plugin_yaml,
     get_validation_definition_from_plugin_definitions,
+    get_plugin_validations_from_definitions,
 )
 
 __all__ = (
-    ValidatorResult.__name__,
-    ValidatorFinding.__name__,
-    ValidatorFindings.__name__,
-    FindingLocation.__name__,
-    FindingSeverity.__name__,
-    get_validation_definition_from_plugin_yaml.__name__,
-    get_validation_definition_from_plugin_definitions.__name__,
+    "ValidatorResult",
```

Comment:
Why is it better to use the string literal rather than the `*.__name__` reference? With the `*.__name__` piece, if it gets changed in the code, we don't have to update here. But would if we used the string literals? If I am thinking about that correctly, anyway.

Reply:
If you're using the Pyright LSP server (I think this is what VSCode uses in the Python extension) it handles renames appropriately and this is the preferred Pythonic way to do it from what I've seen in other projects, docs, warnings on the `__all__` symbol from my editor, etc.

Given that it's handled appropriately (presumably by the main Python LSP servers, at least), I'd recommend we move to this style.

Reply:
The official MS Python plugin supports Pylance or Jedi for the LSP implementation.
<img width="358" alt="Screenshot 2023-09-07 at 8 21 25 PM" src="https://github.com/DevOps-MBSE/AaC/assets/60155541/a5fc2f33-39c6-4d5e-b430-75249ef4c16b">

Reply:
I guess that's what happens when I make assumptions about VSCode. 🙃 That said, while I generally don't like Microsoft stuff - pyright seems to work pretty well.

Reply:
Haha, love that my sanity check spurred a small tooling discussion. 

So, for Cameron's initial comment back, agree the string literal is the general Pythonic way I have seen and previously used. But since this PR was swapping over from the `*.__name__` I took that to mean that those components had possibly previously been often to change. Which I suppose is a good reason to swap to the literal if they don't change often in truth. 

`__all__` can be finicky with linters from what I have seen in the past. It is a useful exposure management component for your packages but can be funky to judge proper inclusion sometimes. Definitely still an area I am a bit shaky in my understanding. 

To Alex's comment, agree that standardizing on implementation and tooling would be useful so that it is easier to provide support to each other when we encounter issues. Also, would help for clarifying what support we can provide to users for issues they have.

Reply:
Using the same Python LSP server could be beneficial but "standardizing on implementation and tooling" is not something I'm sure we should do (depending on how "standard" you want it to be - i.e. do you mean everyone has to use the same editor/IDE?).

For the record, I just tested renaming the `OperationCancelled` class in VSCode using Jedi and Pylance and it also updates the string in `__all__` so even standardizing LSP server doesn't seem totally necessary unless there are *other* reasons to do so.

Reply:
It's definitely something I am back and forth on because there are pretty even pros and cons to standardizing and to being diverse too. Primarily thinking from it in an onboarding stance of "here's our base tooling configuration" that can be provided to a new dev to get them going.

Reply:
The word "tooling" is extremely general - can you please give some examples or more specific description of which specific tools you would potentially like to standardize?

Reply:
So, don't think it makes sense to "mandate" the IDE that is used, outside of some work with VSCode because of the extension we develop. 

But having a list of "quality of life" type tools could be useful, i.e. LSP server, linters, static analysis, etc. So, I am meaning "tools" more akin to IDE extensions. 

Again, primarily from the onboarding side of things to know what a new dev needs to get installed. But also help limit introducing odd environment behaviors and get the lovely line of "Well it runs on my machine". 

Thankfully there isn't a lot of additional environment installations that are part of being able to use AaC like I've had on previous projects to potentially introduce rogue behavior. So, it might not really be necessary to hard analyze and work at coming up with a "tooling" list. And we have the environment configuration files for GitPod and Emacs that help address this in some ways (at least I know GitPod does). 

Based on the pieces of my comment above, I may just be overthinking this and worrying about unlikely impacts. Which is quite plausible.

Reply:
I'd say Python-specific tools like that (minus the Python LSP server) are included in the dev dependencies in `setup.py` already. And VSCode extension dev tools should be in the `package.json` file. If we find more tools we need to have people use, we should just add them there, IMO. That said, I don't see an action that needs to be taken, now, with respect to this.